components/src/types/index.ts

```diff
@@ -38,3 +40,48 @@ export type OptionalTitle = AllOrNone<{
   titleId: string
 }>
 export type IconProps = React.SVGProps<SVGSVGElement> & OptionalTitle
+
+export type WithColor = {
+  /** The base color of the component */
+  color?: Hue | 'background' | 'accent'
+  /** The styling of the color of the component */
+  colorScheme?: 'primary' | 'secondary' | 'gradient' | 'transparent' | 'text'
+}
+
+export type WithAlert = {
+  alert?: 'error' | 'warning' | 'info'
+}
+
+type LegacyTypography =
+  | 'small'
+  | 'large'
+  | 'extraLarge'
+  | 'label'
+  | 'labelHeading'
+
+export type Typography =
+  | 'Heading/H1'
+  | 'Heading/H2'
+  | 'Heading/H3'
+  | 'Heading/H4'
+  | 'Large/XL Normal'
+  | 'Large/XL Bold'
+  | 'Large/Normal'
+  | 'Large/Bold'
+  | 'Body/Normal'
+  | 'Body/Bold'
+  | 'Small/Normal'
+  | 'Small/Bold'
+  | 'Small/XS Normal'
+  | 'Small/XS Bold'
```

Comment:
directly translating typography names from figma doesn't look right because it's not a normal naming scheme within js, maybe use dashes or camelCase?

Reply:
went with camelCase